pricelist_brand/models/product_pricelist.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductPricelist(models.Model):
+
+    _inherit = "product.pricelist"
+
+    def _compute_price_rule_get_items(
+        self, products_qty_partner, date, uom_id, prod_tmpl_ids, prod_ids, categ_ids
+    ):
+        self.ensure_one()
+        product_tmpls = self.env["product.template"].browse(prod_tmpl_ids)
+        brand_ids = product_tmpls.mapped("product_brand_id").ids
+        # Load all rules
+        self.env["product.pricelist.item"].flush(["price", "currency_id", "company_id"])
+        self.env.cr.execute(
```

Comment:
Unfortunately this query make it impossible to inherit this method correctly from the base module. I would suggest a PR to odoo to refactor this query in different methods so we can add a new pricelist option without overriding the standard code.